### DISCUSSION TITLE
Imperial vs Metic

### DISCUSSION THREAD

**Discussion by bendavidson** [tillsteinbach/VWsFriend#320] [Q&A]
Hi,

I'm guessing VW send all the data in metric but it would be handy to view in imperial - especially the range in miles rather than km.  Is this possible currently?  If not, where might I look in the source to change adjust this?

**Comment by tillsteinbach** [comment]
It is currently not possible. To make it configurable all objects in grafana would need to be there twice, one with km and one with mi. I can put that on the TODO list, but it is quite a lot of work I cannot handle right now.

**Reply by bendavidson** [reply]
Thanks for this - I can imagine it's not a small challenge!

I might make a fork and have a play with this myself to see where I get to.

**Reply by tillsteinbach** [reply]
You can have a look how it is done in TeslaMate. They have found a solution.

**Reply by bendavidson** [reply]
I'm not that familiar with alembic and it's uses, but the first stage in this dev is to create a function on the database.  I've created the alembic migration file to do that, but the reference.db that alembic uses is SQLite, which doesn't have functions so when you run alembic upgrade, it fails.

If you change the url in alembic.ini to the postgre install then the upgrade runs just fine.

How would you recommend handling this?

**Reply by tillsteinbach** [reply]
You can check in the migration that it is only executed on postgres
`if op.get_context().dialect.name == 'postgresql':`

**Comment by bendavidson** [comment]
Good shout, thanks!

**Reply by bendavidson** [reply]
Looks like it might not be so bad as I first though - just had to create a new function in the DB and then adjust the SQL in dashboards...now to make a list of all the ones that need changing!
![image](https://user-images.githubusercontent.com/1968527/188169852-bc6914f5-1265-478a-9af7-0c4ec5df7317.png)

**Comment by bendavidson** [comment]
Perfect, thanks!

Hopefully will have a PR to share later this week.

**Reply by Scoobdriver** [reply]
What’s the easiest way to switch over to this fork, would like imperial, do I just change the image name and re-pull ?

**Comment by vincevega-code** [comment]
Is there any way to convert to miles in a format I can follow please?
